File: jsimple-oauth/src/main/java/jsimple/oauth/services/TimestampServiceImpl.java
package jsimple.oauth.services;

import jsimple.util.PlatformUtils;

import java.util.Random;

/**
 * Implementation of {@link TimestampService} using plain java classes.
 *
 * @author Pablo Fernandez
 */
public class TimestampServiceImpl implements TimestampService {
    /**
     * {@inheritDoc}
     */
    public String getNonce() {
        Long ts = getTs();
        return Long.toString(ts + new Random().nextInt());
    }

    /**
     * {@inheritDoc}
     */
    public String getTimestampInSeconds() {
        return getTs().toString();
    }

    private Long getTs() {
        return PlatformUtils.getCurrentTimeMillis() / 1000;
    }
}

File: jsimple-oauth/c#/jsimple/oauth/services/TimestampServiceImpl.cs
using System;

namespace jsimple.oauth.services
{

	using PlatformUtils = jsimple.util.PlatformUtils;

	/// <summary>
	/// Implementation of <seealso cref="TimestampService"/> using plain java classes.
	/// 
	/// @author Pablo Fernandez
	/// </summary>
	public class TimestampServiceImpl : TimestampService
	{
		/// <summary>
		/// {@inheritDoc}
		/// </summary>
		public virtual string Nonce
		{
			get
			{
				long? ts = Ts;
				return Convert.ToString(ts + (new Random()).Next());
			}
		}

		/// <summary>
		/// {@inheritDoc}
		/// </summary>
		public virtual string TimestampInSeconds
		{
			get
			{
				return Convert.ToString(Ts);
			}
		}

		private long? Ts
		{
			get
			{
				return PlatformUtils.CurrentTimeMillis / 1000;
			}
		}
	}

}